diff --git a/xbmc/windowing/amlogic/CMakeLists.txt b/xbmc/windowing/amlogic/CMakeLists.txt
--- a/xbmc/windowing/amlogic/CMakeLists.txt
+++ b/xbmc/windowing/amlogic/CMakeLists.txt
@@ -12,4 +12,10 @@ if(ENABLE_OPENGLES)
   list(APPEND HEADERS WinSystemAmlogicGLESContext.h)
 endif()

 core_add_library(windowing_Amlogic)
+
+if(TARGET ${APP_NAME_LC}::GBM)
+  target_link_libraries(${CORE_LIBRARY} PUBLIC ${APP_NAME_LC}::GBM)
+else()
+  target_link_libraries(${CORE_LIBRARY} PUBLIC gbm)
+endif()
+
diff --git a/xbmc/windowing/gbm/CMakeLists.txt b/xbmc/windowing/gbm/CMakeLists.txt
--- a/xbmc/windowing/gbm/CMakeLists.txt
+++ b/xbmc/windowing/gbm/CMakeLists.txt
@@ -23,4 +23,10 @@ if(TARGET ${APP_NAME_LC}::OpenGLES)
   list(APPEND HEADERS WinSystemGbmGLESContext.h)
 endif()

 core_add_library(windowing_gbm)
+
+if(TARGET ${APP_NAME_LC}::GBM)
+  target_link_libraries(${CORE_LIBRARY} PUBLIC ${APP_NAME_LC}::GBM)
+else()
+  target_link_libraries(${CORE_LIBRARY} PUBLIC gbm)
+endif()
