diff --git a/cmake/modules/FindBluetooth.cmake b/cmake/modules/FindBluetooth.cmake
index 99f7e80e4d..7e95694e3d 100644
--- a/cmake/modules/FindBluetooth.cmake
+++ b/cmake/modules/FindBluetooth.cmake
@@ -24,6 +24,23 @@ if(NOT TARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME})
   if(${${CMAKE_FIND_PACKAGE_NAME}_SEARCH_NAME}_FOUND)
     add_library(${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} ALIAS PkgConfig::${${CMAKE_FIND_PACKAGE_NAME}_SEARCH_NAME})

+    get_target_property(_ALIASTARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} ALIASED_TARGET)
+    if(_ALIASTARGET)
+      set(LIB_TARGET ${_ALIASTARGET})
+    else()
+      set(LIB_TARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME})
+    endif()
+
+    get_target_property(_bluetooth_link_libs ${LIB_TARGET} INTERFACE_LINK_LIBRARIES)
+    if(NOT _bluetooth_link_libs MATCHES "(^|;)bluetooth($|;)" AND
+       NOT _bluetooth_link_libs MATCHES "(^|;)LIBRARY::bluetooth($|;)" AND
+       NOT _bluetooth_link_libs MATCHES "libbluetooth")
+      find_library(BLUETOOTH_LIBRARY NAMES bluetooth)
+      if(BLUETOOTH_LIBRARY)
+        target_link_libraries(${LIB_TARGET} INTERFACE "${BLUETOOTH_LIBRARY}")
+      endif()
+    endif()
+
     set(${${CMAKE_FIND_PACKAGE_NAME}_MODULE}_COMPILE_DEFINITIONS HAVE_LIBBLUETOOTH)
     ADD_TARGET_COMPILE_DEFINITION()
   endif()
diff --git a/xbmc/network/CMakeLists.txt b/xbmc/network/CMakeLists.txt
index ad301ac7de..ed6c776e2f 100644
--- a/xbmc/network/CMakeLists.txt
+++ b/xbmc/network/CMakeLists.txt
@@ -54,6 +54,10 @@ endif()

 core_add_library(network)

+if(TARGET ${APP_NAME_LC}::Bluetooth)
+  target_link_libraries(${CORE_LIBRARY} PRIVATE ${APP_NAME_LC}::Bluetooth)
+endif()
+
 if(ENABLE_STATIC_LIBS AND ENABLE_UPNP)
   target_link_libraries(${CORE_LIBRARY} PRIVATE upnp)
 endif()
